src/cpu/aarch64/acl_inner_product.hpp

```diff
@@ -48,289 +42,38 @@ struct acl_ip_conf_t {
-                ACL_CHECK_VALID(
-                        arm_compute::NEFullyConnectedLayer::has_opt_impl(
-                                expected_weight_format, &aip.src_tensor_info,
-                                &aip.wei_tensor_info,
-                                aip.with_bias ? &aip.bia_tensor_info : nullptr,
-                                &aip.dst_tensor_info, aip.fc_info,
-                                aip.weights_info));
-                aip.weights_info.set_weight_format(expected_weight_format);
-                block_by = arm_compute::block_by(expected_weight_format);
-                if (weights_md_.dims[inner_dim] % block_by != 0)
-                    return status::unimplemented;
-            }
-
-            const memory_desc_t weights_md_received = weights_md_;
-            acl_utils::reorder_to_weight_format(aip.wei_tensor_info,
-                    weights_md_, expected_weight_format, inner_dim, o_dim,
-                    remaining_dims, {});
-
-            ACL_CHECK_SUPPORT(
-                    (weights_format_kind_received == format_kind::blocked)
-                            && !(dnnl_memory_desc_equal(
-                                    &weights_md_received, &weights_md_)),
-                    "specific blocked format not supported by ACL, use "
-                    "format_kind_t::any to find a supported blocked format for "
-                    "your platform");
-
-            // clang-format off
-
-            // Validate fully connected layer manually to check for return status
-            ACL_CHECK_VALID(arm_compute::NEFullyConnectedLayer::validate(
-                &aip.src_tensor_info,
-                &aip.wei_tensor_info,
-                aip.with_bias ? &aip.bia_tensor_info : nullptr,
-                &aip.dst_tensor_info,
-                aip.fc_info,
-                aip.weights_info));
-            // clang-format on
-
-            return status::success;
-        }
+                engine_t *engine, format_kind_t weights_format_kind_received);
     }; // pd_t
 
+    // constructor
     acl_inner_product_fwd_t(const pd_t *apd) : primitive_t(apd) {}
 
-    status_t create_resource(
-            engine_t *engine, resource_mapper_t &mapper) const override {
-        if (mapper.has_resource(this)) return status::success;
-
-        auto r = utils::make_unique<acl_ip_resource_t>();
-        if (!r) return status::out_of_memory;
-
-        // Configure the resource based on information from primitive descriptor
-        CHECK(r->configure(pd()->aip));
-        mapper.add(this, std::move(r));
-
-        return status::success;
-    }
-
-    status_t execute(const exec_ctx_t &ctx) const override {
-        return execute_forward(ctx);
-    }
+    status_t execute(const exec_ctx_t &ctx) const override;
 
 private:
-    //To guard the const execute_forward, the mutex must be 'mutable'
+    // To guard the const execute_forward, the mutex must be 'mutable'
     mutable std::mutex mtx;
```

Comment:
Wasn't the whole point of introducing stateless API to get rid of resource and this mutex along with that?